weed/s3api/auth_credentials.go

```diff
@@ -259,6 +286,7 @@ func (iam *IdentityAccessManagement) authRequest(r *http.Request, action Action)
 		return identity, s3err.ErrAccessDenied
 	}
 
+	r.Header.Set(s3_constants.AmzAccountId, identity.AccountId)
```

Comment:
is this necessary for anonymous mode?

Reply:
Do you mean anonymous access? There is no need to set it for anonymous access. When the request header reads `X-Amz-Account` is empty, the id of `AccountAnonymous` will be used as accountId
```go
func ensureAccountId(r *http.Request) string {
	id := r.Header.Get(s3_constants.AmzAccountId)
	if len(id) == 0 {
		return AccountAnonymous.Id
	} else {
		return id
	}
}
```

Reply:
> is this necessary for anonymous mode?

There are still several `pr` operation objects that need to use this header. The anonymous user is the anonymous accountId, which should not matter.

Reply:
how about only add this header when the user is not anonymous?